engine/src/mbldc.cpp

```diff
@@ -908,11 +908,21 @@ uint16_t MCScreenDC::platform_getheight()
 	return device_getheight() / logicaltoscreenscale();
 }
 
+extern bool MCSystemUpdateMouseCoords(int &x, int &y);
 void MCScreenDC::platform_querymouse(int16_t &r_x, int16_t &r_y)
 {
-	// IM-2014-03-03: [[ Bug 11836 ]] Mouse loc and window position now stored in logical coords
-	r_x = m_mouse_x + m_window_left;
-	r_y = m_mouse_y + m_window_top;
+    int t_x,t_y;
+    if (!MCSystemUpdateMouseCoords(t_x,t_y))
+    {
+        // IM-2014-03-03: [[ Bug 11836 ]] Mouse loc and window position now stored in logical coords
+        r_x = m_mouse_x + m_window_left;
+        r_y = m_mouse_y + m_window_top;
+    }
+    else
+    {
+        r_x = t_x;
+        r_y = t_y;
+    }
 }
```

Comment:
@livecodepanos I think there's an issue with this approach - we want mousex & mousey to represent the x & y coordinates of the last touch event. With the logic as implemented once the scroll is over, those values will revert to what they were before the scroll began.

 I suggest doing away with MCSystemUpdateMouseCoords (the name is misleading) and instead updating MCScreenDC::m_mouse_x and MCScreenDC::m_mouse_y in mbldc.h by adding an update method to MCScreenDC that accepts the new coords and calling that from the platform-specific scroller implementations

Reply:
Thanks @livecodeian.

I am setting this to WIP